### PR TITLE
Add Fleet/Agent 8.6.0 release notes

### LINKED ISSUE #2449: [REQUEST]: Create Fleet/Agent 8.6.0 release notes
### Description

We need to add release notes for the upcoming release of 8.6.0. There is a tool for generating Elastic Agent release notes from changelog fragments, but there will be some manual manipulation required to provide a consolidated view of the release notes.

### Resources

See https://github.com/elastic/observability-docs/pull/2450#issue-1473608069 for links to resources to help finalize this work.

### Collaboration

Other (please describe below in the Point of contact section).

Writing team will work with dev teams to finalize the release notes content.

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #2449

Link to preview: https://observability-docs_2450.docs-preview.app.elstc.co/guide/en/fleet/master/release-notes-8.6.0.html

Reviewers:

- [ ] Please verify the "details" and "impact" sections. Some changelog fragments didn't contain sufficient detail, so I winged it.
- [ ] There was only one changelog fragment for Fleet Server.  It's not really a feature and feels like internals, so I don't think I should add it to the release notes, but let me know if my assumption is wrong: https://github.com/elastic/fleet-server/pull/2079

# Archived notes (please ignore everything below this heading)

Notes for finalizing this PR:

* The dev team has developed a tool to make it easier to maintain changelogs and generate release notes, but it only works for Elastic Agent.
* Assuming we want to provide a single place for users to view Elastic Agent, Fleet, and Fleet Server relnotes, we still need to assemble the content manually.
* I've provided some boilerplate text to make this easier. You may need to modify the boilerplate text and/or the agent changelog entries to get them to work. I feel we can change the style in 8.6, if we want.

## Resources

The source content for the release notes will come from the following places:

- [x]  Elastic Agent changelog fragments. The dev team will generate the changelog for 8.6.0 from existing fragments. Reach out to @endorama or @cmacknz about a week before the release. Note that there is a related issue open here to add some fragments: https://github.com/elastic/elastic-agent/issues/1687
- [x]  Fleet Server changelog fragments. You'll need to look at the changelog fragments in the `fleet-server` repo and incorporate the list of changes manually. Make sure you look at the fragments for the latest BC (basically `https://github.com/elastic/fleet-server/tree/commithash/changelog/fragments` where `commithash` is the hash associated with the release artifact. (Ask dev if you're not sure.)
- [x]  Fleet release notes in Kibana (see https://github.com/elastic/kibana/pull/145768). You'll need to copy any Fleet-related content from each section for the Kibana release notes and possibly modify attribute references to make it work in the Fleet/Agent docs. (Or we could decide not to cover Fleet and tell users to refer to the Kibana docs. Might be awkward for existing users, tho.)

Here are some related PRs and resources:
* [Getting started docs about the elastic-agent-changelog-tool](https://github.com/elastic/elastic-agent-changelog-tool/blob/main/docs/getting-started.md) - you probably don't need this info if dev generates the asciidoc file
* [Discussion about 8.6.0 release notes process](https://github.com/elastic/observability-docs/issues/703)

## Reviewers
When it's ready, this PR should be reviewed by @cmacknz and @jen-huang (or someone they designate)

### CODE REVIEW

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (pierrehilbert) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:102]: The wording has changed for this one:
`Collect logs from sub-processes via stdout and stderr and write them to a single, unified Elastic Agent log file.`

**Comment by pierrehilbert** [comment]
Some changes are missing:

- New control protocol between the Elastic Agent and its subprocesses.
- Internal directory structure change
- All binaries for every supported integration will be bundled in the Elastic Agent by default.

More details here: https://github.com/elastic/elastic-agent/pull/2076/files

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (cmacknz) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:84]:
```suggestion
Be aware that the live query results shown in {kib} may be delayed by up to 5 minutes.
```

We can be more specific about the impact, we know that the results will be up to 5 minutes late because it is tied to the agent checkin period with the same duration.

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (cmacknz) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:117]:
```suggestion
* Health Status: {agent} now indicates detailed status information for each sub-process and input type {fleet-server-pull}1747[#1747] {agent-issue}100[#100]
```

We haven't defined component or unit anywhere so let's avoid use clearer language.

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (cmacknz) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:123]: We should include https://github.com/elastic/fleet-server/issues/2205 as a bug fix.

The changelog entry can be something like "Fix synchronization bug that can lead to Elasticsearch being flooded by requests to /.fleet-actions/_fleet/_fleet_search."

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (cmacknz) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:124]:
```suggestion
* Only show {fleet}-managed data streams on data streams list page {kibana-pull}143300[#143300]
* Fix synchronization bug in {fleet-server} that can lead to Elasticsearch being flooded by requests to /.fleet-actions/_fleet/_fleet_search {fleet-server-pull}2205[#2206].
```

We should include https://github.com/elastic/fleet-server/issues/2205 as a bug fix.

Comment (dedemorton) [reply docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:124]: Added a couple of minor edits to this suggestion before applying.

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:128]:
```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:128]:
```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:128]:
```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:128]:
```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220) {agent-pull}1867[#1867]
```

**Comment by dedemorton** [comment]
>Some changes are missing:

OK, I will add them. I thought they were in the category of changes you did not want to include because they didn't show up in the asciidoc file. Sorry about that....

File: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:108]: @pierrehilbert I've added the missing changelog items here. However, the change to the internal directory structure does not seem like a "feature" to me. Should it be covered under breaking changes instead?

Comment (dedemorton) [reply docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:108]: At this point, we should get this content merged and do follow-up PRs if additional changes are required. I just need approval....

Comment (pierrehilbert) [reply docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:108]: I wanted to put it into "upgrade" kind but seems to be not working that's why I changed to "feature". But yes "breaking-change" makes probably more sense.

Comment (cmacknz) [reply docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc:108]: The internal directory structure change shouldn't be a breaking change, in that the user shouldn't need to take any action to account for it. Agent should keep working. It is definitely notable though.

**Review by cmacknz — APPROVED** [review]
I'm approving this to unblock the release notes going up. DeDe make any changes you feel are appropriate in the current PR, and we can use a follow up PR to adjust it further if needed.

**Comment by dikshachauhan-qasource** [comment]
Hi @dedemorton 

As per current changes suggested in PR, we are able to find notable changes on the Fleet release notes link.

Link validated: https://www.elastic.co/guide/en/fleet/8.6/release-notes-8.6.0.html

Thanks!!
QAS

**Pull request merged**
